Plugin/Model/ResourceModel/Catalog/GetValidCategoryPagesForSitemap.php

```diff
@@ -0,0 +1,259 @@
+
+        if (!$store) {
+            return [];
+        }
+
+        $connection = $subject->getConnection();
+
+        /**
+         * Get show_in_sitemap attribute details
+         */
+        $showInSitemapAttributeDetails = $this->getAttributeDetails('show_in_sitemap');
+
+        // phpcs:disable
+        $this->_select = $connection->select()->from(
+            $subject->getMainTable()
+        )->where(
+            $subject->getIdFieldName() . '=?',
+            $store->getRootCategoryId()
+        );
+        // phpcs:enable
+        $categoryRow = $connection->fetchRow($this->_select);
+
+        if (!$categoryRow) {
+            return [];
+        }
+
+        $this->_select = $connection->select()->from(
+            ['e' => $subject->getMainTable()],
+            [$subject->getIdFieldName(), 'updated_at']
+        )->joinLeft(
+            ['url_rewrite' => $subject->getTable('url_rewrite')],
+            'e.entity_id = url_rewrite.entity_id AND url_rewrite.is_autogenerated = 1'
+            . $connection->quoteInto(' AND url_rewrite.store_id = ?', $store->getId())
+            . $connection->quoteInto(' AND url_rewrite.entity_type = ?', CategoryUrlRewriteGenerator::ENTITY_TYPE),
+            ['url' => 'request_path']
+        )->joinLeft(
+            ['x' => $showInSitemapAttributeDetails['table']],
+            'x.attribute_id = '. $showInSitemapAttributeDetails['attribute_id'] .' AND x.row_id = e.row_id',
+            ['xml' => 'x.value']
+        )->where(
+            'e.path LIKE ?',
+            $categoryRow['path'] . '/%'
+        )->where('x.value = ?', 1);
+
+        $this->addFilter($subject, $storeId, 'is_active', 1);
+
+        $query = $connection->query($this->_select);
+        while ($row = $query->fetch()) {
+            $category = $this->prepareCategory($subject, $row);
+            $categories[$category->getId()] = $category;
+        }
+
+        return $categories;
+    }
+
+    /**
+     * Get attribute details by attribute code
+     *
+     * @param string $attributeCode
+     * @return array
+     * @throws LocalizedException
+     */
+    public function getAttributeDetails(string $attributeCode): array
+    {
+        if (!isset($this->_attributesCache[$attributeCode])) {
+            $attribute = $this->categoryResource->getAttribute($attributeCode);
+
+            $this->_attributesCache[$attributeCode] = [
+                'entity_type_id' => $attribute->getEntityTypeId(),
+                'attribute_id' => $attribute->getId(),
+                'table' => $attribute->getBackend()->getTable(),
+                'is_global' => $attribute->getIsGlobal(),
+                'backend_type' => $attribute->getBackendType(),
+            ];
+        }
+        $attribute = $this->_attributesCache[$attributeCode];
+
+        return $attribute;
+    }
+
+    /**
+     * Prepare category
+     *
+     * Overide due to getCollection function required this function, also this function is protected visibility
```

Comment:
Typo - override.
Also, it says the function is protected, but it's private? Do you mean it was protected in the core class?

Reply:
Original core class visibility is protected.

`    protected function _prepareCategory(array $categoryRow)
    {
        $category = new \Magento\Framework\DataObject();
        $category->setId($categoryRow[$this->getIdFieldName()]);
        $categoryUrl = !empty($categoryRow['url']) ? $categoryRow['url'] : 'catalog/category/view/id/' .
            $category->getId();
        $category->setUrl($categoryUrl);
        $category->setUpdatedAt($categoryRow['updated_at']);
        return $category;
    }`

Reply:
@aligent-lturner Fixed the typo issue for override. Please check.